README-Docker.md

```diff
@@ -0,0 +1,248 @@
+```Shell
+GITHUB_STABLE_DIFFUSION="https://github.com/santisbon/stable-diffusion.git"
+
+cd ~
+git clone $GITHUB_STABLE_DIFFUSION
+
+cd stable-diffusion/docker-build
+chmod +x entrypoint.sh
+
+wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-aarch64.sh -O anaconda.sh && chmod +x anaconda.sh
+```
+
+Build the Docker image. Give it any tag ```-t``` that you want.  
+Tip: Check that your shell session has the env variable set (above) with ```echo $GITHUB_STABLE_DIFFUSION```.  
+```condaarch``` will restrict the conda environment to the right architecture when installing packages. It can take on: ```linux-64```, ```osx-64```, ```osx-arm64```.
+```Shell
+docker build -t santisbon/stable-diffusion \
+--platform linux/arm64 \
+--build-arg condaarch="osx-arm64" \
+--build-arg gsd=$GITHUB_STABLE_DIFFUSION \
+--build-arg sdreq="requirements-linux-arm64.txt" \
+.
+```
+
+Run a container using your built image e.g.
+```Shell
+docker run -it \
+--rm \
+--platform linux/arm64 \
+--name stable-diffusion \
+--hostname stable-diffusion \
+--mount source=my-vol,target=/data \
+--expose 9090 \
+--publish 9090:9090 \
+santisbon/stable-diffusion
+```
+Tip: Make sure you've created the Docker volume (above)
+
+## Option B - Directly on Apple silicon
+For Mac M1/M2. Read more about [Metal Performance Shaders (MPS) framework](https://developer.apple.com/documentation/metalperformanceshaders).
+
+### Prerequisites
+Install the latest versions of macOS, [Homebrew](https://brew.sh/), [Python](https://gist.github.com/santisbon/2165fd1c9aaa1f7974f424535d3756f7#python), and [Git](https://gist.github.com/santisbon/2165fd1c9aaa1f7974f424535d3756f7#git).  
+
+```Shell
+brew install cmake protobuf rust
+brew install --cask miniconda
+conda init zsh && source ~/.zshrc # or bash and .bashrc
+```
+
+### Setup
+
+Set the fork you want to use.
+```Shell
+GITHUB_STABLE_DIFFUSION="https://github.com/santisbon/stable-diffusion.git"
+
+git clone $GITHUB_STABLE_DIFFUSION
+cd stable-diffusion
+mkdir -p models/ldm/stable-diffusion-v1/
+```
+
+When the pip3 path exists, it will ```w```ipe.  
+
+The subdir env variable restricts conda to only use ARM packages while creating the env (M1/M2 is ARM-based). You could also ```conda install nomkl``` but setting the environment appropriately is cleaner.  
+
+```conda config``` will write to the active environment's (```ldm```) configuration file and set ```subdir``` to the desired value permanently.
+```Shell
+PATH_TO_CKPT="$HOME/Downloads"  # or wherever you saved sd-v1-4.ckpt
+ln -s "$PATH_TO_CKPT/sd-v1-4.ckpt" models/ldm/stable-diffusion-v1/model.ckpt
+
+PIP_EXISTS_ACTION="w"
+CONDA_SUBDIR="osx-arm64"
+conda env create -f environment-mac.yaml && conda activate ldm
+conda config --env --set subdir "osx-arm64"
+```
+
+You can verify you're in the virtual environment by looking at which executable you're getting:
+```Shell
+type python3
+```
+
+**Face Restoration and Upscaling**  
```

Comment:
Just before we released 1.14, there was a change in how Real-ESRGAN and GFPGAN are installed. esrgan and its dependencies are now in requirements.txt and environment.yaml (for all platforms), and the GFPGAN repository is pulled from a fork I made for it. The rationale for this is that GFPGAN contains a bug that distorts the color of reconstructed images and its maintainers did not respond to repeated PRs from myself and others.

So now GFPGAN and its model are located underneath stable-diffusion in the src directory. Your instructions should be modified accordingly (and checked that everything still works as expected!)

Reply:
Sounds good! Looks like the new Build workflow is giving me some trouble on my fork so I need to look into that first and then I'll update the doc.

Reply:
Probably related to this change so I'll ask before opening an Issue. I got this behavior when using both lstein/stable-diffusion and my synced fork on Mac M2.

On the latest version on main, a model is no longer being downloaded into ./src/gfpgan/experiments/pretrained_models/GFPGANv1.3.pth throwing an Exception so GFPGAN is not initialized and the -G option doesn't seem to have an effect. It does generate the image, though.

I checked to see if I was missing any new setup instructions on the docs but no. Should I go ahead and open an Issue?